### ISSUE TITLE
IDP example missing POST response encoding

### ISSUE THREAD

**Issue by rmharris157** [IdentityPython/pysaml2#714]
## Code Version
hash 1aeae3a

## Expected Behavior
When running the idp.py test, SAMLResponse should be parseable, and it's coming through as bytes, not a string, so the string lookup can't find it and it fails...

```
IDP listening on 0.0.0.0:8088                                                                                                                      KeyError('SAMLRequest')                                                                                                                            Traceback (most recent call last):                                                                                                                   File "./idp.py", line 452, in post                                                                                                                   _key = saml_msg["key"]                                                                                                                         KeyError: 'key'
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/server.py", line 1283, in communicate                                               req.respond()                                                                                                                                    File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/server.py", line 1073, in respond                                                   self.server.gateway(self).respond()                                                                                                              File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/wsgi.py", line 140, in respond
    response = self.req.server.wsgi_app(self.env, self.start_response)
  File "./idp.py", line 1077, in application
    return func()
  File "./idp.py", line 458, in post
    saml_msg["SAMLRequest"], BINDING_HTTP_POST
KeyError: 'SAMLRequest'
^C^CTraceback (most recent call last):
  File "./idp.py", line 1159, in <module>
    SRV.start()
  File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/server.py", line 1819, in start
    self.serve()
  File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/server.py", line 1798, in serve
    self.tick()
  File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/server.py", line 2038, in tick
    conn = self._connections.get_conn()
  File "/home/ubuntu/venv.test/lib/python3.8/site-packages/cheroot/connections.py", line 137, in get_conn
    in self._selector.select(timeout=0.01)
  File "/home/ubuntu/.pyenv/versions/3.8.5/lib/python3.8/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt
```

## Current Behavior
The received data is not encoded back to a string and they response key lookup fails.

## Possible Solution
in idp.py:
     def unpack_post(self):
    ```_dict = parse_qs(get_post(self.environ))```
becomes
    ```_dict = parse_qs(get_post(self.environ).decode('utf-8'))```

## Steps to Reproduce
1. Used default idp_conf.py and idp settings with minimal reconfiguration. Using AWS SSO as SP.

**Comment by peppelinux** [comment]
I think that this depends on the way you implemented your webserver/sgi I/O.
Try to use djangosaml2idp or uniauth as test and see if this error persist, I don't know but something tells me that idp test still rely on cherrypy

**Comment by peppelinux** [comment]
Hi @rmharris157 would you enjoy to purpose a PR for this, it seems to me that you've build a kind of patch. I'm wondering if there would be need to make some unit test even for example project!

**Comment by rmharris157** [comment]
Yes, I was using the built-in cherrypy infra as part of my testing. 
The patch is literally just the possible solution change above. I can submit a PR if you want.

**Comment by peppelinux** [comment]
@c00kiemon5ter could you incourage this task?
Probably the tests are faulty, what do you think about this PR

**Comment by peppelinux** [comment]
@rmharris157 I don't know if this patch could be not retrocompatible with all the supported py versions.
Would you like to try this

````
post = get_post(self.environ)
qs = post if isinstance(post, str) else post.decode('utf-8') 
_dict = parse_qs(post)
````

**Comment by rmharris157** [comment]
If you mean:

```
post = get_post(self.environ)
qs = post if isinstance(post, str) else post.decode('utf-8') 
_dict = parse_qs(qs)     # You have parse_qs(post) above.
``` 

Then 👍 .

**Comment by peppelinux** [comment]
Yes, fill free to make your fork and do a PR for this.
Please create a separate branch and others and any that @ivan can tell us

**Comment by c00kiemon5ter** [comment]
Fixed by d62a10c473f64e389d2d26958c83c83daafe38fd
